FPIS/Views/AddDepartment.cs

```diff
@@ -0,0 +1,155 @@
+                labelDepartmentName.Text = "Department name is required!";
+                _isDataValid = false;
+                return;
+            }
+            CheckIfDepartmentExist(departmentName);
+            Regex regex = new(nameRegexPattern);
+
+            if (!regex.IsMatch(departmentName))
+            {
+                labelDepartmentName.ForeColor = System.Drawing.Color.Red;
+                labelDepartmentName.Text = "Department name cannot have numbers.";
+                _isDataValid = false;
+                return;
+            }
+        }
+        private void CheckIfDepartmentExist(string departmentName)
+        {
+            AppDbContext dbContext = new();
+            var department = dbContext.Departments.FirstOrDefault(department => department.DepartmentName == departmentName);
+            dbContext.Dispose();
+
+            if (department != null)
+            {
+                _isDataValid = false;
+                labelDepartmentName.ForeColor = System.Drawing.Color.Red;
+                labelDepartmentName.Text = "Department already Created.";
+                return;
+            }
+        }
+        private void ValidateInputs()
+        {
+            ValidateDepartmentName();
+         
+        }
+
+        private void CreateDepartment()
+        {
+            btnAddDepartment.Enabled = false;
+
+            Department department = new()
+            {
+                DepartmentName = materialTextBoxDepartmentName.Text, 
+            };
+            AppDbContext dbContext = new();
+            try
+            {
+                
+                dbContext.Departments.Add(department);
+                dbContext.SaveChanges();
+
+                MessageBox.Show(
+                    $"\"{department.DepartmentName}\" Department created successfully.",
+                    "Success",
+                    MessageBoxButtons.OK,
+                    MessageBoxIcon.Information
+                );
+
+            }
+            catch (Exception ex)
+            {
+                Console.WriteLine($"Error creating Department: {ex}");
+                MessageBox.Show(ex.ToString());
+            }
+            finally
+            {
+                btnAddDepartment.Enabled = true;
+                materialTextBoxDepartmentName.Text = "";
+                dbContext.Dispose();
+            }
+        }
+
+        private void ConfirmNewDepartment()
+        {
+            DialogResult dialogResult = MaterialMessageBox.Show(
+                $"Do you want to create \"{departmentName}\" as a new Department?",
+                "Confirm",
+                MessageBoxButtons.YesNo,
+                MessageBoxIcon.Question
+                );
+
+            if (dialogResult == DialogResult.Yes)
+            {
+                CreateDepartment();
+            }
+            else if (dialogResult == DialogResult.No)
+            {
+                materialTextBoxDepartmentName.Text = "";
+            }
```

Comment:
Please don't delete the entire word the user typed.

Let's say I had to type `Production` and winded up typing `Productio` so I would click No hoping I could add the `n` only to find everything cleared

Reply:
Alright,
I've resolved it